### PR TITLE
Separated test config for enabled species to its own file

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
With this change, it makes it easier to toggle species specifically for the tests. This allows us to only enable the following and save space in the testing rom:
- 281/538 (52.2%) families
- 998/1524 (65.5%) species

Also
- Removed 2 leftover assumes.

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
The idea is to have species off by default that aren't used in tests?

File: include/config/test/species_enabled.h [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+#define P_REGIONAL_FORMS                 TRUE
+#define P_ALOLAN_FORMS                   P_REGIONAL_FORMS
+#define P_GALARIAN_FORMS                 P_REGIONAL_FORMS
+#define P_HISUIAN_FORMS                  P_REGIONAL_FORMS
+#define P_PALDEAN_FORMS                  P_REGIONAL_FORMS
+
+// Big groups of forms that aren't always desired when choosing families.
+#define P_PIKACHU_EXTRA_FORMS            TRUE
+#define P_COSPLAY_PIKACHU_FORMS          P_PIKACHU_EXTRA_FORMS
+#define P_CAP_PIKACHU_FORMS              P_PIKACHU_EXTRA_FORMS
```
Comment (AlexOn1ine) [inline include/config/test/species_enabled.h:37]: those can probably be set to false. I doubt any is used in tests

**Comment by mrgriffin** [comment]
So if I write a test which uses a different species, what happens? I get [`INVALID` due to `IsSpeciesEnabled` failing?](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/test/test_runner_battle.c#L1528)? I suppose that's not too bad. Should update the message to point at `includo/config/test/species_enabled.h`, especially for downstream users who may have written their own tests using species that are now disabled?

Ah... Does this invalidate the tests in [`test/species.c`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/test/species.c), and the species-related tests in [`test/text.c`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/test/text.c)? With species disabled, I guess the tests in `species.c` will skip over those species and not verify the data, and the tests in `text.c` will skip over those species and not check that the names are short enough for all the UIs?

I think maybe it was Egg who suggested that if tests need more space they could disable all the cries?

**Comment by AlexOn1ine** [comment]
> I think maybe it was Egg who suggested that if tests need more space they could disable all the cries?

I believe me and Agustin suggested it

**Comment by AlexOn1ine** [comment]
> > > I think maybe it was Egg who suggested that if tests need more space they could disable all the cries?
> > 
> > 
> > I believe me and Agustin suggested it
> 
> Oops, sorry! 😅

All good!

**Comment by AsparagusEduardo** [comment]
> The idea is to have species off by default that aren't used in tests?
> 
> Maybe we should make a list with mons that should preferably be used in tests? So that we avoid turning on other species.

Yep, that's the idea :)

> So if I write a test which uses a different species, what happens? I get [`INVALID` due to `IsSpeciesEnabled` failing](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/test/test_runner_battle.c?rgh-link-date=2024-08-07T14%3A06%3A39Z#L1528)? Should update the message to point at `include/config/test/species_enabled.h`, especially for downstream users who may have written their own tests using species that are now disabled?

It ends with `ASSUMPTION_FAILED`, not `INVALID`, but sure, I can do add that notice :)

> Ah... Does this invalidate the tests in [`test/species.c`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/test/species.c?rgh-link-date=2024-08-07T14%3A06%3A39Z), and the species-related tests in [`test/text.c`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/test/text.c?rgh-link-date=2024-08-07T14%3A06%3A39Z)? With species disabled, I guess the tests in `species.c` will skip over those species and not verify the data, and the tests in `text.c` will skip over those species and not check that the names are short enough for all the UIs?

Huh. Weirdly enough, I didn't get an assume fail for those tests. Let me get you back on that.

> I think maybe it was ~~Egg~~ Alex and Agustin who suggested that if tests need more space they could disable all the cries?

I can add the `#undef P_CRIES_ENABLED #define P_CRIES_ENABLED FALSE` on this if you guys wish. Only potential issue I can think of is that issues move animations that use cries could be covered by them.

**Comment by AsparagusEduardo** [comment]
- Disabled Pikachu extra forms.
- Added a test version of `config/pokemon.h` to disable cries in tests.
- Added message to point where species are enabled in tests.

> Ah... Does this invalidate the tests in [`test/species.c`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/test/species.c?rgh-link-date=2024-08-07T14%3A06%3A39Z), and the species-related tests in [`test/text.c`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/test/text.c?rgh-link-date=2024-08-07T14%3A06%3A39Z)? With species disabled, I guess the tests in `species.c` will skip over those species and not verify the data, and the tests in `text.c` will skip over those species and not check that the names are short enough for all the UIs?

@mrgriffin I misread earlier what you meant. I don't believe it invalidates it. If the species is disabled, the data is not read, and therefore shouldn't cause issues 🙂

**Comment by mrgriffin** [comment]
> @mrgriffin I misread earlier what you meant. I don't believe it invalidates it. If the species is disabled, the data is not read, and therefore shouldn't cause issues 🙂

Correct me if I'm wrong, but I think the species/text tests will check only the species which are explicitly enabled (i.e. used in battle tests)? So for example, if Venusaur wasn't used in any tests, then it wouldn't have any data, and so we couldn't check that its name fits on the UI or that its forms are valid?

**Comment by AsparagusEduardo** [comment]
> Correct me if I'm wrong, but I think the species/text tests will check only the species which are explicitly enabled (i.e. used in battle tests)? So for example, if Venusaur wasn't used in any battle tests, then it wouldn't have any data, and so we couldn't check that its name fits on the UI or that its forms are valid?

...oh! Now I see what you mean...
Then we can't really disable any species in that case 😅

Let me know if I should change this PR so it just disables cries and be done with it xD

**Comment by AlexOn1ine** [comment]
> > Correct me if I'm wrong, but I think the species/text tests will check only the species which are explicitly enabled (i.e. used in battle tests)? So for example, if Venusaur wasn't used in any battle tests, then it wouldn't have any data, and so we couldn't check that its name fits on the UI or that its forms are valid?
> 
> ...oh! Now I see what you mean... Then we can't really disable any species in that case 😅
> 
> Let me know if I should change this PR so it just disables cries and be done with it xD

I wouldn't disable cries now. We are still far from reaching our limit and once that happens we can decide on a path.

**Comment by AsparagusEduardo** [comment]
> I wouldn't disable cries now. We are still far from reaching our limit and once that happens we can decide on a path.

Probably not our limit, but I figured that users that merged stuff like music expansion probably made space by removing species, so it's a precaution for them and make sure that they csn still run tests.

**Comment by AlexOn1ine** [comment]
> > I wouldn't disable cries now. We are still far from reaching our limit and once that happens we can decide on a path.
> 
> Probably not our limit, but I figured that users that merged stuff like music expansion probably made space by removing species, so it's a precaution for them and make sure that they csn still run tests.

Very valid point. 

In this case would there be a reason to have a placeholder cry that just simulates the cry being played?

**Comment by AsparagusEduardo** [comment]
> In this case would there be a reason to have a placeholder cry that just simulates the cry being played?

Sure, I can do that. Do we use Wobb's

**Comment by AlexOn1ine** [comment]
> > In this case would there be a reason to have a placeholder cry that just simulates the cry being played?
> 
> Sure, I can do that. Do we use Wobb's?

That's fitting, ye.

**Comment by AsparagusEduardo** [comment]
I wasn't able to do much headway on this, as I had issues with the includes. I'll probably go back to this in the future, but for now I'll close the PR.

**Pull request closed without merging**